addon/components/frost-numeric-select.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Component definition for the numeric selection field component
+ */
+import layout from '../templates/components/frost-numeric-select'
+import Component from './frost-component'
+import Ember from 'ember'
+import {PropTypes} from 'ember-prop-types'
+import computed, {readOnly} from 'ember-computed-decorators'
+const {isPresent} = Ember
+
+export default Component.extend({
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
+  layout,
+
+  // == PropTypes =============================================================
+
+  propTypes: {    // options
+    align: PropTypes.string,
+    isHookEmbedded: PropTypes.bool,
+    receivedHook: PropTypes.string,
+    tabindex: PropTypes.number,
+    type: PropTypes.string,
+    isArrowVisible: PropTypes.bool,
+    isAllowInputNegative: PropTypes.bool,
+    isAllowUseDownButtonToNegative: PropTypes.bool,
+    showError: PropTypes.bool,
+    errorMessage: PropTypes.string,
+    class: PropTypes.string,
+
+    // Setting these as part of establishing an inital value
+    autocapitalize: PropTypes.string,
+    autofocus: PropTypes.bool,
+    autocorrect: PropTypes.string,
+    form: PropTypes.string,
+    maxlength: PropTypes.number,
+    placeholder: PropTypes.string,
+    required: PropTypes.bool,
+    selectionDirection: PropTypes.string,
+    spellcheck: PropTypes.bool,
+    value: PropTypes.string,
+    title: PropTypes.string
+
+    // state
+  },
+
+  getDefaultProps () {
+    return {
+      // options
+      align: 'left',
+      autocapitalize: 'off',
+      autocorrect: 'off',
+      autofocus: false,
+      isArrowVisible: false,
+      isAllowInputNegative: true,
+      isAllowUseDownButtonToNegative:false,
+      showError: false,
+      errorMessage: 'Cannot input invalid character',
+      isHookEmbedded: false,
+      required: false,
+      spellcheck: false,
+      tabindex: 0,
+      type: 'text'
+    }
+  },
+
+  // == Functions =============================================================
+
+  _isNumberKey (e) {
+    let decimalExist = this.get('value') && this.get('value').indexOf('.') >=0
+    let charCode = (e.which) ? e.which : e.keyCode
```

Comment:
Is 0 a valid key code? Probably not, so urn should be safe, just remember, if it's anything falsy it'll move to the other side of the ||, including `0`, `''`, or `false`. I know we've had at least one bug where a `false` value wasn't being handled properly because of a check like that. Just a reminder to be cautious.